File: Backend/controller/kas/wallet/klayTransaction.js
const caver = require('caver-js');
const wallet = require('../../../service/kas/wallet');

const klayTransaction = async function (req, res) {
    try {
        // req.body.value의 경우 string이든 int든 관계없음
        const peb = caver.utils.convertToPeb(req.body.value, 'KLAY'); // KLAY를 Peb으로 환산
        const hexpeb = caver.utils.numberToHex(peb);
        const transaction = await wallet.kasKlayTransaction(req.body.from, hexpeb, req.body.to);

        if (transaction['statusCode'] == 200) {
            res.status(200);
            res.json({statusCode: 200, transactionHash: transaction['data']['transactionHash']});
        } else {
            const errorCode = transaction['data']['code'];
            res.status(transaction['statusCode']);

            switch (errorCode) {
                case 1000000:
                    res.json({statusCode: transaction['statusCode'], msg: "송신자와 수신자의 주소가 같습니다."});
                    break;                
                case 1061609:
                    res.json({statusCode: transaction['statusCode'], msg: "송신자 또는 수신자 Klaytn 주소가 올바르지 않습니다. 다시 확인해주세요."});
                    break;
                case 1061615:
                    res.json({statusCode: transaction['statusCode'], msg: "보내는 KLAY 양이 너무 많습니다."});
                    break;
                case 1065001:
                    res.json({statusCode: transaction['statusCode'], msg: "송신자의 잔액이 부족합니다."});
                    break;
                default:
                    res.json({statusCode: transaction['statusCode'], msg: "KLAY 전송에 실패했습니다. 잠시 후 다시 시도해주세요."});
                    break;
            }
        }
    } catch (e) {
        console.error(e);
    }
}

module.exports = klayTransaction;

File: Etiquette/lib/Providers/KAS/Wallet/klay_transaction.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:Etiquette/Models/Settings.dart';

// KLAY 트랜젝션 수행
Future<Map<String, dynamic>> klayTransaction(String from, String value, String to) async {
  try {
    const url = "$SERVER_IP/kas/wallet/klayTransaction";
    final res = await http.post(Uri.parse(url), body: {
      "from": from,
      "value": value,
      "to": to,
    }); // 트랜젝션 수행 시 송신자에게 0.00525 KLAY의 수수료 발생
    Map<String, dynamic> data = json.decode(res.body);
    return data;
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}